sips/sip-034/sip-034.md

```diff
@@ -0,0 +1,188 @@
+dimensions:  the number or indexed reads ("read count"), the number of bytes
+read ("read length"), the number of indexed writes ("write count"), the number
+of bytes written ("write length"), and the amount of CPU used ("runtime").  Each
+miner is granted a budget of each measured resource to consume over the duration
+of their tenure.  Each Clarity operation consumes one or more of these
+resources.  A Nakamoto block is only valid if the additional resources its
+transactions consume do not exceed the tenure's limits on each of these
+resources.
+
+The cost measurements for each Clarity operation were chosen to
+ensure that a Stacks node would be able to reprocess the tenure much more
+quickly than it is produced.  If this were not the case, then Stacks nodes would
+be unable to catch up to the chain tip.  This is particularly dangerous for
+Stacks signers, which may go offline for mundane reasons; signers must be able
+to quickly catch up to the rest of the signer network.
+
+An alternative approach considered to this SIP was to produce a more accurate
+measurement of how much of each resource each Clarity operation consumes.  The
+current model is pessimistic, and assumes a worst-case resource consumption.  It
+also does not consider the size of the _value_ of the data being processed; it
+instead consideres the _worst-case size_ allowed by the data's _type_.
+We believe that the approach in this SIP is superior building a more accurate
+model (but does not negate a new model's usefulness) because it allows
+signers to determine whether or not to allocate more resources to an ongoing
+tenure based on their real-world observation of resource consumption when
+evaluating block proposals.  As long as the signers observe that resource
+consumption in each dimension is below an acceptable threshold, such that a
+crashed signer could quickly catch back up, then the signers can safely extend
+individual resource budgets regardless of the current model's pessimism
+This in turn allows signers to grow the chain's tenure budgets in response
+to acquiring better harder, or running upgraded nodes with better
+block-processing algorithms.
+
+A future SIP may increase the granularity of tenure extensions.  For example,
+the signers may be able to choose the amount of each resource to extend.
+This SIP does not call for this in order to speed its implementation.
+Furthermore, signers have the option to enforce a more granular resource
+consumption rate without a hard fork -- they announce to miners their target
+resource consumption rates, and reject miner blocks that consume resources at
+higher rate than their target rate.
+
+# Backwards Compatibility
+
+Per the Specification section above, the wire formats for `TenureChange`
+transactions are backwards compatible with the existing wire format.  All that
+has been added are new values to represent the kind of tenure extension
+requested.
+
+# Activation
```

Comment:
If activation criteria are redundant because this is a rider to SIP-033, that should be explicitly stated with some criteria or process for clarity.